src/gluonts/dataset/dataframe.py

```diff
@@ -175,6 +182,33 @@ def from_long_dataframe(
         return cls(dataframes=dict(list(dataframe.groupby(item_id))), **kwargs)
 
 
+def series_to_dataframe(
+    series: Union[pd.Series, List[pd.Series], Dict[str, pd.Series]]
+) -> Union[pd.DataFrame, List[pd.DataFrame], Dict[str, pd.DataFrame]]:
+    def to_df(series):
+        assert isinstance(
+            series.index, DatetimeIndexOpsMixin
+        ), "series index has to be a DatetimeIndex."
+        return series.to_frame(name="target")
+
+    if isinstance(series, list):
+        return list(map(to_df, series))
+    elif isinstance(series, dict):
+        return {k: to_df(s) for k, s in series.items()}
+    return to_df(series)
+
+
+def is_series(series: Any) -> bool:
```

Comment:
This looks a case for `TypeGuard`

Reply:
Thanks for the comments @jaheba ! 
I didn't get this one. Do you mean this: https://typeguard.readthedocs.io/ ?
I wouldn't know who to use `TypeGuard` here. I am using this function for the dispatch to `series_to_dataframe` only, not to check for wrong types.

Reply:
I was referring to: https://peps.python.org/pep-0647/

But maybe it's actually not that useful here.